llvm/include/llvm/Demangle/ItaniumDemangle.h

```diff
@@ -5737,6 +5737,7 @@ Node *AbstractManglingParser<Derived, Alloc>::parseTemplateParamDecl(
   }
 
   if (consumeIf("Tk")) {
+    ScopedOverride<bool> SaveInConstraintExpr(InConstraintExpr, true);
```

Comment:
Could we rename `InConstraintExpr` to something that matches what we're trying to do better?
The main problem seems to be that it is not just `<constraint-expr>` that may have problems resolving the names of template parameters due to lack of mapping, but also the template arguments that preceded it.

```
<template-args> ::= I <template-arg>* [Q <requires-clause expr>] E
```

Something like `HasIncompleteTemplateParameterTracking` would make the code more readable.

Reply:
Thanks, renamed.